### DISCUSSION TITLE
GitLab comment length not truncated

### DISCUSSION THREAD

**Discussion by henry-fn** [renovatebot/renovate#30508] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

Essentially https://github.com/renovatebot/renovate/issues/7775 has come back after some refactors. I think that we need a call to smartTruncate around here

https://github.com/renovatebot/renovate/blob/ad7b3ceda800812356b17d7183ced251cd279fc3/lib/modules/platform/gitlab/index.ts#L1405

So you could add a line like `body = smartTruncate(body, maxBodyLength());`

`smartTruncate` is already called in `ensureIssue` but not in `ensureComment`. I think this was just an oversight but it does cause renovate to sometimes crash for us in mysterious ways to our internal users until we inspect the underlying renovate logs. 

### Logs (if relevant)

<details><summary>Logs</summary>

```
"message": "Response code 400 (Bad Request)",
             "stack": "HTTPError: Response code 400 (Bad Request)\n    at Request.<anonymous> (/usr/lib/node_modules/renovate/node_modules/got/dist/source/as-promise/index.js:118:42)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)",
             "options": {
               "headers": {
                 "user-agent": "RenovateBot/37.440.7 (https://github.com/renovatebot/renovate)",
                 "accept": "application/json",
                 "authorization": "***********",
                 "content-type": "application/json",
                 "content-length": "1001123",
                 "accept-encoding": "gzip, deflate, br"
               },
               "url": "REDACTED/merge_requests/3149/notes",
               "hostType": "gitlab",
               "username": "",
               "password": "",
               "method": "POST",
               "http2": false
             },
             "response": {
               "statusCode": 400,
               "statusMessage": "Bad Request",
               "body": {
                 "message": "400 Bad request - Note {:note=>[\"is too long (maximum is 1000000 characters)\"]}"
               },
               "headers": {...

```

</details>

**Comment by RahulGautamSingh** [comment]
We should truncate it before comparing the comment body, else we update the comment each time.

PR welcome for this, if you have time. 🙌

**Reply by RahulGautamSingh** [reply, marked as answer]
Created issue: https://github.com/renovatebot/renovate/issues/30551
